ingest-app/src/cmr/ingest/services/granule_bulk_update/opendap/echo10.clj

```diff
@@ -72,11 +73,37 @@
         updated-resources (conj other-resources cloud-resource on-prem-resource)]
     (remove nil? updated-resources)))
 
+(defn- appended-online-resources
+  "Append urls only if there are no conflicting URLs."
+  [online-resources url-map]
+  (let [opendap-resources (filter #(is-opendap? (:type %)) online-resources)
+        other-resources (remove #(is-opendap? (:type %)) online-resources)
+        current-opendap-urls (map :url opendap-resources)
+
+        _ (when-let [cloud-url (first (filter opendap-util/cloud-url? current-opendap-urls))]
```

Comment:
put these validations in a function in `opendap-util`, so you can call it in here and in `umm-g` as well.